javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -690,7 +684,9 @@ public T get(int index) {
         if ((index < 0) || (index >= length())) {
             throw new IndexOutOfBoundsException("get(" + index + ")");
         } else {
-            return trie.get(index + indexShift).get();
+            final T[] leaf = trie.getLeaf(index);
+            final int leafIndex = lastDigit(trie.offset() + index);
+            return leaf[leafIndex];
```

Comment:
I could put this particular logic inside `BMT`, but it wouldn't scale.
I think that would hinder reusability, as `leaf` and `index` will be used by the optimized iterator also.
In addition you can't primitivize a getter (it would have to return an `Object`), but you can cast an array and get the value out without boxing.
If you still think I should extract it, I will.